snapcraft/internal/meta/snap.py

```diff
@@ -140,7 +156,13 @@ def _validate_required_keys(self) -> None:
         if missing_keys:
             raise errors.MissingSnapcraftYamlKeysError(keys=missing_keys)
 
-    def validate(self) -> None:
+        required_grade = self._get_required_grade()
```

Comment:
Mind just bunching this into `validate_grade()`?

Reply:
sure, public method or private though?

Reply:
Ah, private (whatever is consistent)